src/us/kbase/workspace/test/database/mongo/MongoWorkspaceDBTest.java

```diff
@@ -623,6 +661,93 @@ public Exception failAddDataToObjects(
 		}
 	}
 	
+	@Test
+	public void addDataToObjectsBackendScaling() throws Exception {
+		/* The scaling factor is really just an implementation detail and doesn't affect the
+		 * data returned from the method, so we really shouldn't care about testing it
+		 * other than it doesn't cause failures. The only noticeable effect should be that
+		 * a higher scaling factor should result in a faster return, which is not something
+		 * that should be in automated tests as it's heavily dependent on the environment.
+		 * Furthermore, in automated tests like this where network speed is not an issue
+		 * little to no benefit will be seen from parallelizing the requests as IO should be
+		 * local disk bound.
+		 * 
+		 * Hence, this test just tries different scaling factors and prints out the speed
+		 * for informational purposes. Any rigorous speed testing should take place outside the
+		 * context of unit or even integration tests.
+		 */
```

Comment:
All that said, even in this unrealistic test, where I'd expect the impact of threading to be the least, it has a pretty significant impact. This is the result of the print statement at the end of this test:
```
{1=4619, 2=2207, 5=1103, 10=848, 50=832, 100=876, 500=1232}
```